xds/src/main/java/io/grpc/xds/AbstractXdsClient.java

```diff
@@ -64,13 +64,20 @@
       "type.googleapis.com/envoy.api.v2.RouteConfiguration";
   private static final String ADS_TYPE_URL_RDS =
       "type.googleapis.com/envoy.config.route.v3.RouteConfiguration";
-  private static final String ADS_TYPE_URL_CDS_V2 = "type.googleapis.com/envoy.api.v2.Cluster";
+  @VisibleForTesting
+  static final String ADS_TYPE_URL_CDS_V2 = "type.googleapis.com/envoy.api.v2.Cluster";
   private static final String ADS_TYPE_URL_CDS =
       "type.googleapis.com/envoy.config.cluster.v3.Cluster";
   private static final String ADS_TYPE_URL_EDS_V2 =
       "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment";
   private static final String ADS_TYPE_URL_EDS =
       "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
+  @VisibleForTesting
+  static final String TYPE_URL_UPSTREAM_TLS_CONTEXT =
```

Comment:
These two typeUrl should be defined in `ClientXdsClient` as they are used for client's specific business logic. 

The abstract layer only decode xDS resource blobs to `List<Any>`, the remaining processing should be done in the  business logic layer.

Reply:
There is an existing example for HttpConnectionManager:
https://github.com/grpc/grpc-java/blob/90a3873a131a3b9169ee286b21c65fb4f0b39680/xds/src/main/java/io/grpc/xds/ClientXdsClient.java#L71-L76

Reply:
fixed